emergence_lib/src/player_interaction/details.rs

```diff
@@ -0,0 +1,285 @@
+        pub(crate) active_recipe: Option<RecipeId>,
+
+        /// The state of the ongoing crafting process.
+        pub(crate) state: CraftingState,
+
+        /// The time remaining to finish crafting.
+        pub(crate) timer: Timer,
+    }
+
+    impl Display for CraftingDetails {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            let input_inventory = &self.input_inventory;
+            let output_inventory = &self.output_inventory;
+            let recipe_id = &self.active_recipe;
+            let crafting_state = &self.state;
+            let time_remaining = self.timer.remaining_secs();
+            let total_duration = self.timer.duration().as_secs_f32();
+
+            write!(
+                f,
+                "Input: {input_inventory}
+                Output: {output_inventory}
+                Recipe ID: {recipe_id:?}
+                {crafting_state}: {time_remaining:.2} s / {total_duration:.2} s
+                "
+            )
+        }
+    }
+}
+
+mod unit {
+    use bevy::ecs::{prelude::*, query::WorldQuery};
+    use std::fmt::Display;
+
+    use crate::{
+        organisms::units::{
+            behavior::{CurrentAction, Goal},
+            item_interaction::HeldItem,
+            UnitId,
+        },
+        simulation::geometry::TilePos,
+    };
+
+    /// Data needed to populate [`StructureDetails`].
+    #[derive(WorldQuery)]
+    pub(super) struct UnitDetailsQuery {
+        /// The type of unit
+        pub(super) unit_id: &'static UnitId,
+        /// The current location
+        pub(super) tile_pos: &'static TilePos,
+        /// What's being carried
+        pub(super) held_item: &'static HeldItem,
+        /// What this unit is trying to acheive
+        pub(super) goal: &'static Goal,
+        /// What is currently being done
+        pub(super) action: &'static CurrentAction,
+    }
+
+    /// Detailed info about a given unit.
+    #[derive(Debug)]
+    pub(crate) struct UnitDetails {
+        /// The type of unit
+        pub(super) unit_id: UnitId,
+        /// The current location
+        pub(super) tile_pos: TilePos,
+        /// What's being carried
+        pub(super) held_item: HeldItem,
+        /// What this unit is trying to acheive
+        pub(super) goal: Goal,
+        /// What is currently being done
+        pub(super) action: CurrentAction,
+    }
+
+    impl Display for UnitDetails {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            let unit_id = &self.unit_id;
+            let tile_pos = &self.tile_pos;
+            let held_item = &self.held_item;
+            let goal = &self.goal;
+            let action = &self.action;
+
+            write!(
+                f,
+                "
+            Unit type: {unit_id}
+            Tile: {tile_pos}
+            Holding: {held_item}
+            Goal: {goal}
+            Action: {action}
+            "
```

Comment:
This includes the indent in the string, I'd strongly prefer if we used a margin for the text instead if we want this for the design.

For the string definition you can either unindent it or add `\n\` at the end of every line.